utils/helpers.js

```diff
@@ -53,20 +53,55 @@ const getIPFSGateway = () => {
 }
 
 /**
- * Performs feature detection on XRHand and XRMediaBinding to determine if user is on Oculus Quest.
- * As of 10/15/21, only Oculus Browser has implemented the WebXR Hand Input Module and WebXR Layers API.
- * @returns true if XRHand and XRMediaBinding are not null, else false
+ * Performs feature detection and a UA check to determine if user is using Oculus Browser.
+ * @returns an object indicating whether there is a match and the associated confidence level.
  */
-const isOculusQuest = () => {
-  return (window.XRHand != null && window.XRMediaBinding != null)
+const checkOculusBrowser = () => {
+  // As of 5/26/22, only Oculus Browser has implemented the WebXR Hand Input Module and WebXR Layers API.
+  const featureDetect = (window.XRHand != null && window.XRMediaBinding != null);
+  const uaCheck = navigator.userAgent.includes('OculusBrowser');
+  const confidence = featureDetect && uaCheck ? 'full' : 
+                     featureDetect || uaCheck ? 'partial' : 
+                     'none';
+  return { match: confidence !== 'none', confidence: confidence }
+}
+
+/**
+ * Performs feature detection and a UA check to determine if user is using Wolvic.
+ * @returns an object indicating whether there is a match and the associated confidence level.
+ */
+const checkWolvicBrowser = () => {
+  // While Wolvic is still shipping with a Gecko backend, this feature detect should hold true.
+  // Once versions with different backends start showing up in the wild, this will need revisiting.
+  const featureDetect = (window.XRHand == null && window.XRMediaBinding == null);
+  const uaCheck = navigator.userAgent.includes('Mobile VR') && !navigator.userAgent.includes('OculusBrowser');
+  const confidence = featureDetect && uaCheck ? 'full' : 
+                     featureDetect || uaCheck ? 'partial' : 
+                     'none';
+  return { match: confidence !== 'none', confidence: confidence }
+}
+
+/**
+ * Performs feature detection and a UA check to determine if user is using Pico's browser.
+ * @returns an object indicating whether there is a match and the associated confidence level.
+ */
+ const checkPicoBrowser = () => {
+  // Pico's internal browser is a Chromium fork and seems to expose some WebXR AR modules,
+  // so perform an isSessionSupported() check for immersive-vr and immersive-ar.
+  const featureDetect = (navigator.xr.isSessionSupported('immersive-vr') && navigator.xr.isSessionSupported('immersive-ar'));
```

Comment:
Is the exposing of `immersive-ar` a permanent feature of the pico browser? If not, this might cause issues going forward, if this were to be removed.

Reply:
That's true yeah. I'll check if it has hand tracking support and if so we can use that as a feature detect